pkg/dmaasbackup/periodic_backup.go

```diff
@@ -16,58 +16,82 @@ package dmaasbackup
 import (
 	"time"
 
-	"github.com/mayadata-io/dmaas-operator/pkg/apis/mayadata.io/v1alpha1"
 	"github.com/pkg/errors"
 	"github.com/robfig/cron"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mayadata-io/dmaas-operator/pkg/apis/mayadata.io/v1alpha1"
 )
 
 func (d *dmaasBackup) processPeriodicConfigSchedule(obj *v1alpha1.DMaaSBackup) error {
 	d.logger.Debug("Processing fullbackup")
+	defer d.logger.Debug("Processing fullbackup completed")
+
+	// We may have queued empty velero schedule entry with name for schedule creation
+	// check if such entry exist
+	dummySchedule := getDummyVeleroSchedule(obj)
+	if dummySchedule != nil {
+		// we have queued dummy schedule for full backup
+		// let's create new schedule using name from dummy schedule
+		// and update it
+		newSchedule, err := d.createScheduleUsingName(obj, dummySchedule.ScheduleName)
+		if err != nil {
+			// TODO : check for already created schedule
+			d.logger.WithError(err).Errorf("failed to create new schedule")
+			return err
+		}
+		updateDummyVeleroSchedule(obj, dummySchedule, newSchedule)
+
+		lastSchedule := getLastVeleroSchedule(obj)
+		if lastSchedule == nil {
+			// there is not last created schedule
+			return nil
+		}
+
+		err = d.scheduleClient.Delete(
```

Comment:
what happens to any pending backups on this schedule? will they proceed?

Reply:
yes.

Reply:
better to delete them or add a TODO if this can be covered later
and hope cstor can handle full backup or incrementals with multiple schedules going on